src/Illuminate/Http/JsonResponse.php

```diff
@@ -83,4 +83,15 @@ public function setEncodingOptions($options)
 
         return $this->setData($this->getData());
     }
+
+    /**
+     * Checks if error happened during json_encode.
+     *
+     * @param  int  $jsonLastError
+     * @return bool
+     */
+    protected function hasJsonError($jsonLastError)
+    {
+        return $jsonLastError !== JSON_ERROR_NONE && ($jsonLastError !== JSON_ERROR_UNSUPPORTED_TYPE || ! ($this->encodingOptions & JSON_PARTIAL_OUTPUT_ON_ERROR));
```

Comment:
This line is pretty hard to read. Is it possible to break it down into something simpler?